docs/container-register-disks.md

```diff
@@ -0,0 +1,247 @@
+the vm image as a block device is what we provide in the standardized KubeVirt
+base wrapper container.
+
+When virt-handler sees a VM is placed on the local node with the
+**ContainerRegistryDisk:v1alpha** disk type defined, virt-handler injects the
+necessary configuration into the domain xml file required to connect to the
+block device served up by the wrapper container.
+
+The **v1** part of the ContainerRegistryDisk disk type represents the standard
+used during the virt-handler disk conversion process. As we gain more
+experience with this feature, we may want to adopt a new standard for how VM
+images are wrapped by a container while maintaining backwards compatibility. 
+
+**Bringing it all together**
+* KubeVirt provides a base container that serves up VM disks as a network
+  block device libvirt can consume
+* Users push their VM disks into the container registry using the base image
+* Users reference VM disks backed by the container registry with the
+  **ContainerRegistryDisk:v1alpha** disk type.
+* The virt-launcher pod has an instance of the wrapper container which serves
+  up the VM disk as an ephemeral block device.
+* virt-handler sees the ContainerRegistryDisk:v1alpha disk type and automatically
+  injects disk configuration into the domain xml that gives the VM access to the
+  block device served up by the wrapper container.
+* Anything the VM writes to disk is destroyed when when the VM is torn down as a
+  result of the virt-launcher pod (which has the wrapper container) being
+  destroyed.
+
+### v1 Base Container Design
+
+**iscsi Base Container**
+
+The **v1** base container will serve up a VM disk as a block device using
+iscsi. Conceptually, this base container will do the equivalent of what the
+images/iscsi-demo-target-tgtd does today. The big difference here is no actual
+VM disk will be present in the base container, only the dependencies and
+entry point script required to serve up a user provided image.
+
+**Recommended qcow2 Usage**
+
+Although we don't have to enforce this, It is recommended that users upload
+their images in qcow2 format.
+
+This is important for two reasons.
+1. VMs with large disks can be stored in the container registry while only
+   consuming the minimal amount of network storage.
+2. The most common union mount filesystems do not act on block storage, but
+   instead copy up on file writes. To avoid the copy up performance penalty,
+   the wrapper container will have its own unique copy of the VM disk. Using
+   qcow2 will reduce the local disk space required. The container image will
+   only need the qcow2 file, while the actual running wrapper container will
+   expand that image to raw format.
+
+Example of the logic the wrapper container will use to expand a qcow2 image to
+a raw image that can be served as a block device. The qcow2 image could be 1gb
+which could expand to a 100gb ephemeral raw file.
+```
+qemu-img convert image.qcow2 image.raw
```

Comment:
Another downside is, like you already said, that the overlay file systems will treat the image as one blob, no  matter if it is a `raw` or a `qcow2` image. Therefore, we don't get the nice delta benefits, when we update our image in the registry.